### PR TITLE
(BKR-410) `confine :except, {}, [host1,host2]` does not work as expected

### PR DESCRIPTION
**Pull request by anodelman**
- add support to confine to all hosts *except* those in provided host
  array

### CODE REVIEW

File: lib/beaker/dsl/structure.rb [hunk 1]
```diff
@@ -175,10 +179,17 @@ def confine(type, criteria, host_array = nil, &block)
...
+            hosts_to_modify = hosts_to_modify - select_hosts(criteria, hosts_to_modify, &block)
+          else
+            # confining to all hosts *except* provided array of hosts
+            hosts_to_modify = hosts - host_array
+          end
         when :to
           if criteria and ( not criteria.empty? )
             hosts_to_modify = select_hosts(criteria, hosts_to_modify, &block)
+          else
+            # confining to only hosts in provided array of hosts
```
Comment (colinPL) [inline lib/beaker/dsl/structure.rb]: This is an empty `else` portion.

Comment (anodelman) [inline lib/beaker/dsl/structure.rb]: Yup, just wanted the comment in there to be super clear what the fall through case is.  Not a coding syntax that you like?

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by anodelman** [comment]
retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Pull request merged**